server/gfmd/user.c

```diff
@@ -2011,18 +2117,49 @@ gfm_server_user_auth_modify(struct peer *peer,
 	if (e == GFARM_ERR_NO_ERROR) {
 
 		if (auth_user_id == NULL || strcmp(auth_user_id, "") == 0) {
-			e = user_auth_id_remove(u, auth_user_id_type_str);
+			bool need_to_update_db = false;
+			e = user_auth_id_remove_internal(u,
+				auth_user_id_type_str, &need_to_update_db);
+
+			if (e == GFARM_ERR_NO_ERROR &&
+				need_to_update_db) {
+
+				struct db_user_auth_remove_arg arg = {
+					username,
+					auth_user_id_type_str
+				};
+				e = db_user_auth_remove(&arg);
```

Comment:
the error of db_user_auth_remove() should be logged by gflog_error(), it's somewhat critical error.